openlr/openlr_sample.cpp

```diff
@@ -0,0 +1,161 @@
+#include "openlr/openlr_sample.hpp"
+
+#include "indexer/index.hpp"
+
+#include "base/string_utils.hpp"
+
+#include <cerrno>
+#include <cstring>
+#include <fstream>
+
+namespace
+{
+void ParseMWMSegments(std::string const & line, uint32_t const lineNumber,
+                      std::vector<openlr::SampleItem::MWMSegment> & segments, Index const & index)
+{
+  std::vector<string> parts;
+  strings::ParseCSVRow(line, '=', parts);
+
+  for (auto const & seg : parts)
+  {
+    std::vector<string> segParts;
+    strings::ParseCSVRow(seg, '-', segParts);
+    ASSERT_EQUAL(segParts.size(), 5, ());
+
+    auto const mwmId = index.GetMwmIdByCountryFile(platform::CountryFile(segParts[0]));
+
+    uint32_t featureIndex;
+    if (!strings::to_uint(segParts[1], featureIndex))
+      MYTHROW(openlr::SamplePoolLoadError, ("Can't parse MWMSegment", seg, "line:", lineNumber));
+
+    uint32_t segId;
+    if (!strings::to_uint(segParts[2], segId))
+      MYTHROW(openlr::SamplePoolLoadError, ("Can't parse MWMSegment", seg, "line:", lineNumber));
+
+    bool const isForward = (segParts[3] == "fwd");
+    double length = 0;
+    if (!strings::to_double(segParts[4], length))
+      MYTHROW(openlr::SamplePoolLoadError, ("Can't parse MWMSegment", seg, "line:", lineNumber));
+
+    segments.push_back({FeatureID(mwmId, featureIndex), segId, isForward, length});
+  }
+}
+
+void ParseSampleItem(std::string const & line, uint32_t const lineNumber, openlr::SampleItem & item,
+                     Index const & index)
+{
+  std::vector<string> parts;
+  strings::ParseCSVRow(line, '\t', parts);
+  ASSERT_GREATER_OR_EQUAL(parts.size(), 2, ());
```

Comment:
Same as above - CHECK or throw an exception, as parts are results of line splitting.